src/ml/neural_net/tf_compute_context.cpp

```diff
@@ -0,0 +1,63 @@
+/* Copyright © 2018 Apple Inc. All rights reserved.
+ *
+ * Use of this source code is governed by a BSD-3-clause license that can
+ * be found in the LICENSE.txt file or at https://opensource.org/licenses/BSD-3-Clause
+*/
+
+#include <iostream>
+#include <vector>
+
+#include <ml/neural_net/tf_compute_context.hpp>
+#include <pybind11/pybind11.h>
+#include <pybind11/numpy.h>
+
+namespace turi {
+namespace neural_net {
+namespace py = pybind11;
+namespace {
+
+std::unique_ptr<compute_context> create_tf_compute_context() {
+  return std::unique_ptr<compute_context>(new tf_compute_context);
+}
+
+// At static-init time, register create_tf_compute_context().
+// TODO: Codify priority levels?
+static auto* tf_registration = new compute_context::registration(
+    /* priority */ 1, &create_tf_compute_context);
```

Comment:
Worth noting in a comment that this priority is intended to make TensorFlow come _after_ MPS (on Macs, where both are available).